src/file.cpp

```diff
@@ -224,6 +229,11 @@ read_file_result read_file(const char *path) {
   posix_fd_file file(fd);
   return read_file(path, file.ref());
 }
+
+read_file_result read_stdin() {
+  posix_fd_file file(STDIN_FILENO);
```

Comment:
Almost-blocking: `posix_fd_file` is the wrong class to use. `posix_fd_file::~posix_fd_file` will close the file descriptor, which we probably don't want to do. Use `posix_fd_file_ref` instead.

(Same comment applies to `windows_handle_file`.)